src/prelude/internals.mo

```diff
@@ -612,9 +612,22 @@ func @timer_helper() : async () {
   ignore (prim "global_timer_set" : Nat64 -> Nat64) exp;
   if (exp == 0) @timers := null;
 
+  var failed : Nat64 = 0;
+  func reinsert(job : () -> async ()) {
+    if (failed == 0) @timers := @prune @timers;
```

Comment:
is this ìf`supposed to fall through to next code?

Reply:
The PQ needs to be pruned of expunged expirations only once.

But this is also the place where we should set the global timer to 1.

Reply:
Done in 321786aee3ccfe910c42b16c02c7c48030c91e9c.